pkg/sanitization/sanitization.go

```diff
@@ -0,0 +1,30 @@
+package sanitization
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	// validKubeNameRegex is the regex to validate a kubernetes resource name
+	validKubeNameRegex = regexp.MustCompile(`[^a-z0-9\-]+`)
+
+	// moreThanOneHyphen is the regex to check if there is more than one hyphen together
+	moreThanOneHyphen = regexp.MustCompile(`-(-+)`)
+)
+
+func SanitizeName(name string) string {
+	name = strings.TrimSpace(name)
+	name = strings.ToLower(name)
+
+	name = validKubeNameRegex.ReplaceAllString(name, "-")
+
+	name = moreThanOneHyphen.ReplaceAllString(name, "-")
+	// trim the repository name for internal use in labels
+	if len(name) > 63 {
```

Comment:
Can we extract 63 to a constant with meaning instead of a magic number? For example, `const maxKubernetesResourceLenght = 63` or similar? I think it is important to indicate the meaning of 63